README.md

```diff
@@ -1,2 +1,175 @@
-# go-libp2p-relay
-Circuit Switching for libp2p
+# go-libp2p-circuit-relay
+> Circuit Switching in libp2p
+
+Table of Contents
+- [Overview](#overview)
+- [Dramatization](#dramatization)
+- [Addressing](#addressing)
+- [/ipfs/relay/circuit stream protocol](#ipfsrelaycircuit-stream-protocol)
+- [Interfaces](#interfaces)
+- [Removing existing relay protocol](#removing-existing-relay-protocol)
+
+
+### Overview
+
+The circuit relay is a means of establishing connectivity between
+libp2p nodes (such as IPFS) that wouldn't otherwise be able to connect to each other.
+
+This helps in situations where nodes are behind NAT or reverse proxies,
+or simply don't support the same transports (e.g. go-ipfs vs. browser-ipfs).
+libp2p already has modules for NAT ([go-libp2p-nat][go-libp2p-nat]),
+but these don't always do the job, just because NAT traversal is a clusterfuck.
+That's why it's useful to have a simple relay protocol.
+
+One node asks a relay node to connect to another node on behalf of the first node.
+The relay node shortcircuits its streams to the two nodes,
+and they are then connected through the relay.
+This connection quaks and behaves just like a direct connection would,
+because it is in fact a direct connection, but not over e.g. a TCP stream,
+but over an existing swarm stream with another peer.
+
+Relayed connections are end-to-end encrypted just like direct connections.
+
+The circuit relay is both a tunneled transport and a mounted swarm protocol.
+The transport is the means to *establishing* and *accepting* connections,
+and the swarm protocol is the means to *relaying* connections.
+
+```
++-------+    /ip4/.../tcp/.../ws/ipfs/QmRelay    +---------+    /ip4/.../tcp/ipfs/QmTwo    +-------+
+| QmOne | <------------------------------------> | QmRelay | <---------------------------> | QmTwo |
++-------+                                        +---------+                               +-------+
+      ^                                            +-----+                                    ^
+      |                                            |     |                                    |
+      |        /libp2p-circuit-relay/QmTwo         |     |                                    |
```

Comment:
yeah, we can short all the libp2p specific transports to only use `p2p`, at the same time we drop `/ipfs/` and use `/p2p/` for addrs